services/121-service/test/payment/do-payment-fsp-nedbank.test.ts

```diff
@@ -0,0 +1,517 @@
+        );
+        expect(exportPayment.nedbankOrderCreateReference).toBeDefined();
+        expect(exportPayment.nedbankPaymentReference).toMatchSnapshot();
+      });
+
+      it('should fail pay-out when debitor account number is missing', async () => {
+        const registrationFailDebitorAccount = {
+          ...registrationNedbank,
+          phoneNumber: NedbankMockNumber.failDebitorAccountIncorrect,
+        };
+        const paymentReferenceIds = [
+          registrationFailDebitorAccount.referenceId,
+        ];
+        await seedIncludedRegistrations(
+          [registrationFailDebitorAccount],
+          programId,
+          accessToken,
+        );
+
+        // Act
+        await doPayment(
+          programId,
+          payment,
+          amount,
+          paymentReferenceIds,
+          accessToken,
+        );
+
+        await waitForPaymentTransactionsToComplete(
+          programId,
+          paymentReferenceIds,
+          accessToken,
+          30_000,
+          [
+            TransactionStatusEnum.success,
+            TransactionStatusEnum.error,
+            TransactionStatusEnum.waiting,
+          ],
+        );
+
+        const getTransactionsBody = (
+          await getTransactions(
+            programId,
+            payment,
+            registrationFailDebitorAccount.referenceId,
+            accessToken,
+          )
+        ).body;
+
+        // Assert
+        expect(getTransactionsBody[0].status).toBe(TransactionStatusEnum.error);
+        expect(getTransactionsBody[0].errorMessage).toMatchSnapshot();
+      });
+
+      it('should fail pay-out when we make a payment with a payment amount of over 5000', async () => {
+        const amountOver6000 = 6000;
+        const paymentReferenceIds = [registrationNedbank.referenceId];
+        await seedIncludedRegistrations(
+          [registrationNedbank],
+          programId,
+          accessToken,
+        );
+
+        // Act
+        await doPayment(
+          programId,
+          payment,
+          amountOver6000,
+          paymentReferenceIds,
+          accessToken,
+        );
+
+        await waitForPaymentTransactionsToComplete(
+          programId,
+          paymentReferenceIds,
+          accessToken,
+          30_000,
+          [
+            TransactionStatusEnum.success,
+            TransactionStatusEnum.error,
+            TransactionStatusEnum.waiting,
+          ],
+        );
+
+        const getTransactionsBody = (
+          await getTransactions(
+            programId,
+            payment,
+            registrationNedbank.referenceId,
+            accessToken,
+          )
+        ).body;
+
+        // Assert
+        expect(getTransactionsBody[0].status).toBe(TransactionStatusEnum.error);
+        expect(getTransactionsBody[0].errorMessage).toMatchSnapshot();
+      });
+
+      it('should set transaction status to error in reconciliation process if phonenumber is incorrect', async () => {
```

Comment:
How can the phone number turn out to be incorrect during reconciliation? We send the phone number when creating the order, so if it was not correct, it would have raised an error then, right?

Reply:
That test scenario is mainly there to check if we handle Nedbank errors in the reconciliation process correctly. This is the only error I got from the Nedbank Api on reconciliation, until the insufficient fund error of the acceptance test. But I did not store that error text/body, so I think using the error is most pratical. 

There could also be other validation on phonenumber happening at Nedbank.